src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,243 @@
+        // then consider the request to have been made the day before.
+        // To compensate, advance startDate by one day.
+        String hour = time.split(":")[0];
+        if (Integer.parseInt(hour) < SERVICE_DAY_START_HOUR) {
+            startDate = startDate.plusDays(1);
+        }
+
+        ZonedDateTime startDateDayBefore = startDate.minusDays(1);
+
+        return (
+            date.equals(startDate.format(dateFormatter)) &&
+                startDate.getHour() >= SERVICE_DAY_START_HOUR
+        ) || (
+            // Trips starting between 12am and 2:59am next day are considered same-day.
+            date.equals(startDateDayBefore.format(dateFormatter)) &&
+                startDateDayBefore.getHour() < SERVICE_DAY_START_HOUR
+        );
+    }
+
+    public static List<Itinerary> getSameDayItineraries(List<Itinerary> itineraries, MonitoredTrip trip, String date)
+        throws URISyntaxException {
+        List<Itinerary> result = new ArrayList<>();
+
+        if (itineraries != null) {
+            for (Itinerary itinerary : itineraries) {
+                // TODO/FIXME: initialize the parameter map in the monitored trip in initializeFromItinerary etc.
+                //   so we don't have to lug the URI exception around.
+                if (isSameDay(itinerary, date, trip.tripTime, DateTimeUtils.getOtpZoneId(), trip.isArriveBy())) {
+                    result.add(itinerary);
+                }
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Checks that, for each query provided, an itinerary exists.
+     * @param labeledQueries a map containing the queries to check, each query having a key or
+     *                       label that will be used in Result.labeledResponses for easy identification.
+     * @return An object with a map of results and summary of itinerary existence.
+     */
+    public static Result checkItineraryExistence(Map<String, String> labeledQueries, boolean tripIsArriveBy) {
+        // TODO: Consider multi-threading?
+        Map<String, OtpResponse> responses = new HashMap<>();
+        boolean allItinerariesExist = true;
+
+        for (Map.Entry<String, String> entry : labeledQueries.entrySet()) {
+            OtpDispatcherResponse response = OtpDispatcher.sendOtpPlanRequest(entry.getValue());
+            responses.put(entry.getKey(), response.getResponse());
+
+            Itinerary sameDayItinerary = response.findItineraryDepartingSameDay(tripIsArriveBy);
+            if (sameDayItinerary == null) allItinerariesExist = false;
+        }
+
+        return new Result(allItinerariesExist, responses);
+    }
+
+    /**
+     * Class to pass the results of the OTP itinerary checks.
+     */
+    public static class Result {
+        /** Whether all itineraries checked exist. */
+        public final boolean allItinerariesExist;
+        /**
+         * A map with the same keys as the input from checkAll,
```

Comment:
What is `checkAll`? I need to read more, but I'm a bit suspect of this labeledQueries approach.

Reply:
`checkAll` determines whether all days of the week are checked. The UI needs to check all days to report which days are not available, whereas `monitoredtrip` PUT/POST only needs to checks trip existence on the days selected for monitoring.